tests/prague/eip3540_eof_v1/test_example_valid_invalid.py

```diff
@@ -0,0 +1,517 @@
+            id="jump_to_opcode_ok",
+        ),
+        pytest.param(
+            # Check that jumps into the middle on an opcode are not allowed
+            Container(
+                name="EOF1I0019",
+                sections=[
+                    Section.Code(
+                        code=Op.RJUMP[3] + Op.RJUMP[2] + Op.RJUMP[-6] + Op.STOP,
+                        code_inputs=0,
+                        code_outputs=NON_RETURNING_SECTION,
+                        max_stack_height=0,
+                    ),
+                    Section.Data("0x0bad60A7"),
+                ],
+            ),
+            "ef0001010004020001000A0400040000800000E00003E00002E0FFFA000bad60A7",
+            EOFException.INVALID_RJUMP_DESTINATION,
+            id="rjump_3_2_m6_fails",
+        ),
+        pytest.param(
+            # Check that jumps into the middle on an opcode are not allowed
+            Container(
+                name="EOF1I0019",
+                sections=[
+                    Section.Code(
+                        code=Op.PUSH1(0)
+                        + Op.PUSH1(0)
+                        + Op.PUSH1(0)
+                        + Op.RJUMPI[3]
+                        + Op.RJUMPI[2]
+                        + Op.RJUMPI[-6]
+                        + Op.STOP,
+                        code_inputs=0,
+                        code_outputs=NON_RETURNING_SECTION,
+                        max_stack_height=3,
+                    ),
+                    Section.Data("0x0bad60A7"),
+                ],
+            ),
+            "ef000101000402000100100400040000800003600060006000E10003E10002E1FFFA000bad60A7",
+            EOFException.INVALID_RJUMP_DESTINATION,
+            id="push1_0_0_0_rjump_3_2_m6_fails",
+        ),
+        pytest.param(
+            # Check that that code that uses removed opcodes fails
+            Container(
+                name="EOF1I0015",
+                sections=[
+                    Section.Code(
+                        code=Op.PUSH1(3) + Op.JUMP + Op.JUMPDEST + Op.STOP,
+                        code_inputs=0,
+                        code_outputs=NON_RETURNING_SECTION,
+                        max_stack_height=1,
+                    ),
+                    Section.Data("0xef"),
+                ],
+            ),
+            "ef0001010004020001000504000100008000016003565B00ef",
+            EOFException.UNDEFINED_INSTRUCTION,
+            id="jump_jumpdest_fails",
+        ),
+    ],
+)
+def test_example_valid_invalid(
+    eof_test: EOFTestFiller,
+    eof_code: Container,
+    expected_hex_bytecode: str,
+    exception: EOFException,
+):
+    """
+    Verify eof container construction and exception
+    """
+    # TODO remove this after Container class implementation is reliable
+    assert bytes(eof_code).hex() == bytes.fromhex(expected_hex_bytecode).hex()
+
+    eof_test(
+        data=eof_code,
+        expect_exception=exception,
+    )
+
+
+def test_new_eof_opcodes(eof_test: EOFTestFiller):
```

Comment:
I think these test cases should go in `src/ethereum_test_tools/tests/test_vm.py`, just like the ones fixed in this PR.